### PR TITLE
Support backslash escape in json path subscripts

### LINKED ISSUE #7499: json_extract doesnt extract keys with double quotes
A JSON column in my source table has the structure
```json
 "parameterValues": {
    "DE:Design and/or Copy - Deliverables Requested": "Presentation Design/Copy",
  },
```
I am unable to JSON_EXTRACT this successfully. Tried

```sql
json_format(json_extract(object_json, '$["parameterValues"]["DE:Design and/or Copy - \"Other\" Deliverables Requested Description"]'))
```
and this

```sql
json_format(json_extract(object_json, '$["parameterValues"]["DE:Design and/or Copy - '"Other"" Deliverables Requested Description"]'))`
```
How do I get around double quotes in keys? I thought backslashes effectively escaped quotes.

### PR DESCRIPTION
**Pull request by findepi**
Fixes #7499

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonPathTokenizer.java [hunk 1]
```diff
@@ -119,16 +122,46 @@ private String matchQuotedSubscript()
...
-        int start = index;
-        while (hasNextCharacter() && peekCharacter() != QUOTE) {
+        StringBuilder token = new StringBuilder();
+        boolean escaped = false;
+
+        while (hasNextCharacter() && (escaped || peekCharacter() != QUOTE)) {
+            if (escaped) {
+                switch (peekCharacter()) {
+                    case 'u':
+                        // Unicode escape, will be handled later
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonPathTokenizer.java]: Dear Reviewer,

I am not sure this is right, I could not find any evidence, but this was expected by an existing test: https://github.com/prestodb/presto/blob/c73359fe2173e01140b7d5f102b286e81c1ae4a8/presto-main/src/test/java/com/facebook/presto/operator/scalar/TestJsonExtract.java#L63 . Is the test right?

**Comment by electrum** [comment]
I think that test is wrong. It should have a single backslash, so the Unicode escape is handled by Java. There's no need to support them inside quoted paths.

**Comment by findepi** [comment]
Thanks. Code updated accordingly.

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**